TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonToByteBufHandler.java

```diff
@@ -154,27 +154,24 @@ private static void writeHeadersAsChunks(ChannelHandlerContext ctx,
     {
         AtomicInteger chunkIdx = new AtomicInteger(headerChunkSizes.size());
         var bufs = headerChunkSizes.stream()
-                .map(i -> ByteBufAllocator.DEFAULT.buffer(chunkIdx.decrementAndGet()==0?maxLastBufferSize:i).retain())
+                .map(i -> ctx.alloc().buffer(chunkIdx.decrementAndGet()==0?maxLastBufferSize:i).retain())
                 .toArray(ByteBuf[]::new);
-        var cbb = ByteBufAllocator.DEFAULT.compositeBuffer(bufs.length);
-        ResourceLeakDetector<CompositeByteBuf> rld =
-                (ResourceLeakDetector<CompositeByteBuf>) ResourceLeakDetectorFactory.instance().newResourceLeakDetector(cbb.getClass());
-        rld.track(cbb);
-        cbb.addComponents(true, bufs);
-        log.debug("cbb.refcnt="+cbb.refCnt());
-        try (var bbos = new ByteBufOutputStream(cbb)) {
-            writeHeadersIntoStream(httpJson, bbos);
-        }
-        log.debug("post write cbb.refcnt="+cbb.refCnt());
-        int debugCounter = 0;
-        for (var bb : bufs) {
-            log.debug("bb[" + (debugCounter) +  "].refcnt=" + bb.refCnt());
-            ctx.fireChannelRead(bb);
-            bb.release();
-            log.debug("Post fire & decrement - bb[" + (debugCounter) +  "].refcnt=" + bb.refCnt());
-            debugCounter++;
+        CompositeByteBuf cbb = null;
+        try {
+            cbb = ctx.alloc().compositeBuffer(bufs.length);
+            cbb.addComponents(true, bufs);
+            log.debug("cbb.refcnt=" + cbb.refCnt());
+            try (var bbos = new ByteBufOutputStream(cbb)) {
+                writeHeadersIntoStream(httpJson, bbos);
+            }
+            for (var bb : bufs) {
+                ctx.fireChannelRead(bb);
+            }
+        } finally {
+            if (cbb != null) {
+                cbb.release();
+            }
```

Comment:
Notice that we use a CompositeByteBuffer for our own purposes with the ByteBufOutputStream.  However, we only fire bytes for the individual components, not the whole composite.  Hence, we have to release the CompositeByteBuf since nobody else is going to.  

Also notice that cbb.release() will release all of its component ByteBufs - WHICH WERE ALSO RELEASED BY line 168.  HOWEVER, this is OK because further up in the file, when we allocate the buffers, we call retain() an extra time.  So - we allocate the components and retain them (2 retains total).  We fireChannelRead for each component AND call release on the parent, resulting in 2 releases - so this evens out.